src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -16613,6 +16613,33 @@ static void G(string s)
             Assert.Equal(true, symbol.Type.IsNullable);
         }
 
+        [Fact]
+        public void InferLocalType_Error()
+        {
+            var source =
+@"using System;
+using System.Collections.Generic;
+static class C
+{
+    static T F<T>(IEnumerable<T> e)
+    {
+        throw new NotImplementedException();
+    }
+    static void G()
+    {
+        var a = new[] { F(a) };
```

Comment:
We should test similar scenario in a top level declaration within a Script.

Reply:
Should probably also test scenarios when ```a``` is reference before its declaration. In a Script too, including mutual dependency between two declarations.

Reply:
Added tests.